pyriemann/datasets/sampling.py

```diff
@@ -89,58 +151,19 @@ def _slice_sampling(ptarget, n_samples, x0, n_burnin=20, thin=10,
 
     rs = check_random_state(random_state)
     w = 1.0  # initial bracket width
-    xt = np.copy(x0)
 
-    n_dim = len(x0)
-    samples = []
     n_samples_total = (n_samples + n_burnin) * thin
 
-    for _ in range(n_samples_total):
-
-        for i in range(n_dim):
-
-            ei = np.zeros(n_dim)
-            ei[i] = 1
-
-            # step 1 : evaluate ptarget(xt)
-            Px = ptarget(xt)
-
-            # step 2 : draw vertical coordinate uprime ~ U(0, ptarget(xt))
-            uprime_i = Px * rs.rand()
-
-            # step 3 : create a horizontal interval (xl_i, xr_i) enclosing xt_i
-            r = rs.rand()
-            xl_i = xt[i] - r * w
-            xr_i = xt[i] + (1-r) * w
-            while ptarget(xt + (xl_i - xt[i]) * ei) > uprime_i:
-                xl_i = xl_i - w
-            while ptarget(xt + (xr_i - xt[i]) * ei) > uprime_i:
-                xr_i = xr_i + w
-
-            # step 4 : loop
-            while True:
-                xprime_i = xl_i + (xr_i - xl_i) * rs.rand()
-                Px = ptarget(xt + (xprime_i - xt[i]) * ei)
-                if Px > uprime_i:
-                    break
-                else:
-                    if xprime_i > xt[i]:
-                        xr_i = xprime_i
-                    else:
-                        xl_i = xprime_i
-
-            # store coordinate i of new sample
-            xt = np.copy(xt)
-            xt[i] = xprime_i
-
-        samples.append(xt)
+    samples = Parallel(n_jobs=n_jobs)(
+        delayed(_slice_one_sample)(ptarget, x0, w, rs)
```

Comment:
this can be dangerous to pass a RandomState like this. You will see that when using different processes the RandomState will be copied so each parallel run will effectively be seeded the same way. I suspect this code lacks diversity of produced samples when run in parallel.

actually read this https://joblib.readthedocs.io/en/latest/auto_examples/parallel_random_state.html maybe with the current loly backend is fine.

can you check that sequential or parallel runs give the same outputs when using a seed?